src/libecalc/presentation/yaml/yaml_types/emitters/yaml_venting_emitter.py

```diff
@@ -1,36 +1,104 @@
 
 
-class YamlEmitterModel(YamlBase):
-    emission_rate: ExpressionType = Field(
+class YamlVentingEmission(YamlBase):
+    name: str = Field(
         ...,
-        title="EMISSION_RATE",
-        description="Emission rate expression [kg/day]",
+        title="NAME",
+        description="Name of emission",
     )
 
+    rate: YamlRate = Field(..., title="RATE", description="The emission rate")
+
 
 class YamlVentingEmitter(YamlBase):
+    class Config:
+        title = "VentingEmitter"
+
+    component_type = ComponentType.VENTING_EMITTER
+
     name: str = Field(
         ...,
         title="NAME",
-        description="Name of the emitter.\n\n$ECALC_DOCS_KEYWORDS_URL/NAME",
+        description="Name of venting emitter",
     )
-    category: YamlTemporalModel[ConsumerUserDefinedCategoryType] = CategoryField(
+
+    emission: YamlVentingEmission = Field(
         ...,
+        title="EMISSION",
+        description="The emission",
     )
-    emission_name: str = Field(
+
+    category: ConsumerUserDefinedCategoryType = CategoryField(
         ...,
-        title="EMISSION_NAME",
-        description="Name of emission",
-    )
-    emitter_model: YamlTemporalModel[YamlEmitterModel] = Field(
-        ..., title="EMITTER_MODEL", description="The emitter model.\n\n$ECALC_DOCS_KEYWORDS_URL/EMITTER_MODEL"
     )
+
+    @property
+    def id(self) -> str:
+        return generate_id(self.name)
+
+    @property
+    def user_defined_category(self):
+        return self.category
+
+    @validator("category", pre=True, always=True)
+    def check_user_defined_category(cls, category, values):
+        """Provide which value and context to make it easier for user to correct wrt mandatory changes."""
+        if category is not None:
+            if category not in list(ConsumerUserDefinedCategoryType):
+                name = ""
+                if values.get("name") is not None:
+                    name = f"with the name {values.get('name')}"
+
+                raise ValueError(
+                    f"CATEGORY: {category} is not allowed for {cls.Config.title} {name}. Valid categories are: {[(consumer_user_defined_category.value) for consumer_user_defined_category in ConsumerUserDefinedCategoryType]}"
+                )
+        return category
+
+    def get_emission_rate(
+        self, variables_map: VariablesMap, regularity: Dict[datetime, Expression]
+    ) -> TimeSeriesStreamDayRate:
+        regularity_evaluated = TemporalExpression.evaluate(
+            temporal_expression=TemporalModel(regularity),
+            variables_map=variables_map,
+        )
+
+        emission_rate = (
+            Expression.setup_from_expression(value=self.emission.rate.value)
+            .evaluate(variables=variables_map.variables, fill_length=len(variables_map.time_vector))
+            .tolist()
+        )
+
+        # emission_rate = Unit.to(self.emission.rate.unit, Unit.KILO_PER_DAY)(emission_rate).tolist()
```

Comment:
remove? or keep?

Reply:
I think it should be removed. But keep the commented line for a little while, to be really sure:

1. There is no core calculations for venting emitters. So I cannot see a need to convert to `kg/d`.
2. The only thing I can think of, is aggregation - but that is done at a later stage, when emissions should be in `t/d`, I think?